### PR TITLE
Feature/bi 13605 upgrade to java 21

### PR DESCRIPTION
**Pull request by JoshFen95**
__Short description outlining key changes/additions__
- update project to java21
- fix security issues where possible(still some issues being brought in by latest version of dependencies, details on ticket)
- add terraform files for ECS deployment
__JIRA Ticket Number__
[BI-13605](https://companieshouse.atlassian.net/browse/BI-13605)
### Type of change

* [ ] Bug fix
* [X] New feature
* [ ] Breaking change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

[BI-13605]: https://companieshouse.atlassian.net/browse/BI-13605?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SteveHicksCH — COMMENTED** [review]
Looks good but we should not merge until we get the new pipeline ready (at this stage you can also delete the meso files)

**Review by SteveHicksCH — COMMENTED** [review]
Can you remove `server.port=${SIC_CODE_API_PORT}` from the application.properties to keep this consistent with other applications

File: terraform/groups/ecs-service/locals.tf
Comment (hwilliams1) [inline terraform/groups/ecs-service/locals.tf:11]: The load balancer priority will need a different value to other applications in the stack that expose an end point. This should be determined by the architect before we merge this.

File: terraform/groups/ecs-service/main.tf
Comment (tobafolaranmi-ch) [inline terraform/groups/ecs-service/main.tf:31]: Please use tag ref `1.0.253`

File: terraform/groups/ecs-service/main.tf
Comment (tobafolaranmi-ch) [inline terraform/groups/ecs-service/main.tf:44]: https://github.com/companieshouse/abridged.accounts.api.ch.gov.uk/pull/517/files

Please use this as a ref for multi albs config

**Comment by tobafolaranmi-ch** [comment]
The terraform directory is missing the version file and the symlink to the vault.tf

File: terraform/groups/ecs-service/locals.tf
Comment (JoshFen95) [reply terraform/groups/ecs-service/locals.tf:11]: Checked the lbs and lb priority in the "search" stack and 21 is free on the load balancer that the sic-code-api will be using. Spoke to toba from platform and is happy with 21 in that case

**Review by SteveHicksCH — APPROVED** [review]

**Review by tobafolaranmi-ch — APPROVED** [review]

**Pull request merged**
